sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -898,20 +899,20 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
           const ListFileSystemsOptions& listFileSystemsOptions)
       {
         Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Get, url);
-        request.GetUrl().AppendQuery(Details::c_QueryResource, "account");
+        request.GetUrl().AppendQueryParameter(Details::c_QueryResource, "account");
         if (listFileSystemsOptions.Prefix.HasValue())
         {
-          request.GetUrl().AppendQuery(
+          request.GetUrl().AppendQueryParameter(
               Details::c_QueryPrefix, listFileSystemsOptions.Prefix.GetValue());
```

Comment:
`Prefix` is a variable rather than string literal like `"account"`, you should encode it.

Reply:
ditto for others